packages/openapi-fetch/src/index.d.ts

```diff
@@ -86,155 +106,107 @@ export default function createClient<Paths extends {}>(
   clientOptions?: ClientOptions,
 ): {
   /** Call a GET endpoint */
-  GET<P extends PathsWithMethod<Paths, "get">>(
+  GET<
+    P extends PathsWithMethod<Paths, "get">,
+    I extends MaybeOptionalInit<Paths[P], "get">,
+  >(
     url: P,
-    ...init: HasRequiredKeys<
-      FetchOptions<FilterKeys<Paths[P], "get">>
-    > extends never
-      ? [(FetchOptions<FilterKeys<Paths[P], "get">> | undefined)?]
-      : [FetchOptions<FilterKeys<Paths[P], "get">>]
+    ...init: I
   ): Promise<
     FetchResponse<
-      "get" extends infer T
-        ? T extends "get"
-          ? T extends keyof Paths[P]
-            ? Paths[P][T]
-            : unknown
-          : never
-        : never
+      Paths[P]["get"],
+      I[0]["parseAs"] extends ParseAs ? I[O]["parseAs"] : "json"
```

Comment:
This seems like there’s a way to avoid this, but I couldn’t think of a way. I needed this in my version for TS to fall back to `json` if `init?.parseAs` was omitted